src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -619,6 +631,126 @@ winrt::fire_and_forget AppHost::_WindowActivated()
     }
 }
 
+void AppHost::_BecomeMonarch(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                             const winrt::Windows::Foundation::IInspectable& /*args*/)
+{
+    _setupGlobalHotkeys();
+}
+
+winrt::fire_and_forget AppHost::_setupGlobalHotkeys()
+{
+    // The hotkey MUST be registered on the main thread. It will fail otherwise!
+    co_await winrt::resume_foreground(_logic.GetRoot().Dispatcher(),
+                                      winrt::Windows::UI::Core::CoreDispatcherPriority::Normal);
+
+    // Remove all the already registered hotkeys before setting up the new ones.
+    _window->UnsetHotkeys(_hotkeys);
+
+    _hotkeyActions = _logic.GlobalHotkeys();
+    _hotkeys.clear();
+    for (const auto& [k, v] : _hotkeyActions)
+    {
+        if (k != nullptr)
+        {
+            _hotkeys.push_back(k);
+        }
+    }
+
+    _window->SetGlobalHotkeys(_hotkeys);
+}
+
+// Method Description:
+// - Called whenever a registered hotkey is pressed. We'll look up the
+//   GlobalSummonArgs for the specified hotkey, then dispatch a call to the
+//   Monarch with the selection information.
+// - If the monarch finds a match for the window name (or no name was provided),
+//   it'll set FoundMatch=true.
+// - If FoundMatch is false, and a name was provided, then we should create a
+//   new window with the given name.
+// Arguments:
+// - hotkeyIndex: the index of the entry in _hotkeys that was pressed.
+// Return Value:
+// - <none>
+void AppHost::_GlobalHotkeyPressed(const long hotkeyIndex)
+{
+    if (hotkeyIndex < 0 || static_cast<size_t>(hotkeyIndex) > _hotkeys.size())
+    {
+        return;
+    }
+    // Lookup the matching keychord
+    Control::KeyChord kc = _hotkeys.at(hotkeyIndex);
+    // Get the stored ActionAndArgs for that chord
+    if (const auto& actionAndArgs{ _hotkeyActions.Lookup(kc) })
+    {
+        if (const auto& summonArgs{ actionAndArgs.Args().try_as<Settings::Model::GlobalSummonArgs>() })
+        {
+            Remoting::SummonWindowSelectionArgs args{ summonArgs.Name() };
+
+            _windowManager.SummonWindow(args);
+            if (args.FoundMatch())
+            {
+                // Excellent, the window was found. We have nothing else to do here.
+            }
+            else
+            {
+                // We should make the window ourselves.
+                _createNewTerminalWindow(summonArgs);

```

Comment:
Worried about the proliferation of blocks of "run wt.exe with these args" code...

Reply:
this is clever/cool though